### PR TITLE
Update module deprecations

### PR DESCRIPTION
**Pull request by jillr**
##### SUMMARY
Update module deprecations.  Switch version to `removed_at_date`.

##### ISSUE TYPE
- Docs Pull Request

### CODE REVIEW

File: plugins/modules/aws_caller_info.py [hunk 1]
```diff
@@ -75,7 +75,7 @@ def main():
         supports_check_mode=True,
     )
     if module._name == 'aws_caller_facts':
-        module.deprecate("The 'aws_caller_facts' module has been renamed to 'aws_caller_info'", version='2.13')
+        module.deprecate("The 'aws_caller_facts' module has been renamed to 'aws_caller_info'", date='2022-06-01', collection_name='amazon.aws')
```
Comment (Qalthos) [inline plugins/modules/aws_caller_info.py:78]: 2.13 should be 2021-12-01
```suggestion
        module.deprecate("The 'aws_caller_facts' module has been renamed to 'aws_caller_info'", date='2021-12-01', collection_name='amazon.aws')
```

Comment (jillr) [reply plugins/modules/aws_caller_info.py]: Not intended, thanks @Qalthos!

File: meta/runtime.yml [hunk 2]
```diff
@@ -50,45 +50,45 @@ plugin_routing:
   modules:
     aws_az_facts:
       deprecation:
-        removal_date: TBD
+        removal_date: 2022-06-01
         warning_text: see plugin documentation for details
     aws_caller_facts:
       deprecation:
-        removal_date: TBD
+        removal_date: 2022-06-01
```
Comment (Qalthos) [inline meta/runtime.yml:57]: This is now out of sync with the module
```suggestion
        removal_date: 2021-12-01
```

Comment (jillr) [reply meta/runtime.yml:57]: *headdesk* thank you

**Pull request merged**